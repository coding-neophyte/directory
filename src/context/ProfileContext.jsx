import React, { createContext } from 'react'
import { getProfile } from '../services/profiles'
import { useContext } from 'react'
import { useState } from 'react';

export const ProfileCtx = createContext();

export const ProfileProvider = ({ children }) => {
    const userProfile = getProfile()
    const [profile, setProfile] = useState( userProfile ? { id: userProfile.id, email: userProfile.email } : {} );

    return <ProfileCtx.Provider value={ { profile, setProfile }}>{children}</ProfileCtx.Provider>
}


export const UseProfile = () => {
    const UseProfile = useContext(ProfileCtx)

    if(UseProfile === undefined){
        throw new Error('invalid')
    }
    return UseProfile;
}


export default { ProfileProvider, UseProfile }
