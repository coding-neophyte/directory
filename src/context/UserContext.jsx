import { useContext, createContext, useState } from "react";
import { getUser } from "../services/users";

const userContext = createContext();

const UserProvider = ({ children }) => {
    const currentUser = getUser();
    const [user, setUser] = useState( currentUser ? { id: currentUser.id, email: currentUser.email } : {} )
    return <userContext.Provider value={{ user, setUser }}>
        {children}
    </userContext.Provider>
}


const useUser = () => {
    const useUser = useContext(userContext);

    if(useUser === undefined){
        throw new Error('invalid provider')
    }
        return useUser;
}

export { userContext, UserProvider, useUser }
