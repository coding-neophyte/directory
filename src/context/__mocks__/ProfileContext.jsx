import React, { createContext } from 'react'
import { useContext } from 'react'
import { useState } from 'react';

export const ProfileCtx = createContext();

export const ProfileProvider = ({ children, mockprofile }) => {
    const [profile, setProfile] = useState( mockprofile ? { id: mockprofile.id, email: mockprofile.email } : {} );

    return <ProfileCtx.Provider value={ { profile, setProfile }}>{children}</ProfileCtx.Provider>
}


export const UseProfile = () => {
    const UseProfile = useContext(ProfileCtx)

    if(UseProfile === undefined){
        throw new Error('invalid')
    }
    return UseProfile;
}


export default { ProfileProvider, UseProfile }
