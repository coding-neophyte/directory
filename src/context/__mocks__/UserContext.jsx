import { useContext, createContext, useState } from "react";

const userContext = createContext();

const UserProvider = ({ children , mockuser }) => {
    const [user, setUser] = useState( mockuser ? { id: mockuser.id, email: mockuser.email } : {} )
    return <userContext.Provider value={{ user, setUser }}>
        {children}
    </userContext.Provider>
}


const useUser = () => {
    const useUser = useContext(userContext);

    if(useUser === undefined){
        throw new Error('invalid provider')
    }
        return useUser;
}

export { userContext, UserProvider, useUser }
