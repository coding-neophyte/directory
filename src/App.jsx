import { BrowserRouter as Router, Switch, Route,  } from 'react-router-dom';
import Home from './views/Home';
import SignIn from './components/SignIn';
import Profile from './views/Profile';
import SignUp from './components/SignUp';
import PrivateRoute from './components/PrivateRoute';
import { UserProvider } from './context/UserContext';
import { ProfileProvider } from './context/ProfileContext';

import EditProfile from './components/EditProfile';
import Header from './components/Header';

export default function App() {
  return (
    <UserProvider>
      <ProfileProvider>
      <Router>
        <Header />
        <Switch>
            <Route exact path='/signin'>
                <SignIn />
            </Route>
            <Route exact path='/signup'>
                <SignUp />
            </Route>
            <Route exact path='/'>
                <Home />
            </Route>
            <PrivateRoute exact path='/profile'>
                <Profile  />
            </PrivateRoute>
            <PrivateRoute exact path='/profile/edit'>
                <EditProfile />
            </PrivateRoute>
        </Switch>
      </Router>
            </ProfileProvider>
      </UserProvider>
    )
}
