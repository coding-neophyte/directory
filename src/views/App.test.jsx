import App from "../App";
import { render } from '@testing-library/react'
import { UserProvider } from "../context/UserContext";
import { ProfileProvider} from '../context/ProfileContext'
import { MemoryRouter } from 'react-router-dom'

jest.mock('../context/UserContext')
jest.mock('../context/ProfileContext')
jest.mock('../services/users')
jest.mock('../services/profiles')

test('redner app and components', () => {
    const { container } = render(
        <UserProvider>
            <ProfileProvider>
    <MemoryRouter initialEntries={['/']}>
            <App />
        </MemoryRouter></ProfileProvider></UserProvider>)





    expect(container).toMatchSnapshot();

})
