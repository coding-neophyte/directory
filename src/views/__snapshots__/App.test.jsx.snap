// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redner app and components 1`] = `
<div>
  <div>
    <header>
      <p>
         
        Not Signed In
         
      </p>
      <a
        href="/signin"
      >
        <button>
          Sign In
        </button>
      </a>
    </header>
  </div>
  <div>
    <h1>
      Welcome to Employee Directory
    </h1>
    <a
      href="/signin"
    >
      Login
    </a>
    <p>
      Or
    </p>
    <a
      href="/signup"
    >
      Create Account
    </a>
  </div>
</div>
`;
